vmdb/app/models/orchestration_template.rb

```diff
@@ -90,6 +93,16 @@ def validate_content(manager = nil)
     "No #{ui_lookup(:model => 'ExtManagementSystem').downcase} is capable to validate the template"
   end
 
+  def validate_format
+    "Cannot validate format because template type is unknown"
+  end
+
+  def save_with_format_validation
+    msg = draft? ? nil : validate_format
+    raise MiqException::MiqParsingError, msg if msg
```

Comment:
Oh, also, how about:

```ruby
def save_with_format_validation
  error_msg = validate_format unless draft?
  raise MiqException::MiqParsingError, error_msg if error_msg
  save!
end
```

Something about the explicit `nil` assignment keeps catching my eye...